### PR TITLE
nvme-cli: Add new package

### PR DESCRIPTION
**Pull request by lu-zero**
Maintainer: me
Compile tested: aarch64
Run tested: bpi-r3
Description: lib + cli for diagnosing/manipulating nvme

### CODE REVIEW

File: libs/libnvme/Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=libnvme
+PKG_VERSION:=1.5
+PKG_RELEASE:=1
+
+PKG_SOURCE:=libnvme-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/linux-nvme/$(PKG_NAME)/archive/refs/tags/v$(PKG_VERSION)
+PKG_BUILD_DIR:=$(BUILD_DIR)/libnvme-$(PKG_VERSION)
+PKG_HASH:=f73ba1edde059b2d5e7c1048ad4f895e6047bff241c94b34a7aff5894779d086
```
Comment (1715173329) [inline libs/libnvme/Makefile:15]:
```suggestion
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/linux-nvme/libnvme/archive/refs/tags/v$(PKG_VERSION)
PKG_HASH:=f73ba1edde059b2d5e7c1048ad4f895e6047bff241c94b34a7aff5894779d086
```

PKG_BUILD_DIR defaults to `$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)`, no need to specify

File: libs/libnvme/Makefile [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2023 Luca Barbato
+
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
```
Comment (1715173329) [inline libs/libnvme/Makefile:6]: seems unused?

File: libs/libnvme/Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+PKG_NAME:=libnvme
+PKG_VERSION:=1.5
+PKG_RELEASE:=1
+
+PKG_SOURCE:=libnvme-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/linux-nvme/$(PKG_NAME)/archive/refs/tags/v$(PKG_VERSION)
+PKG_BUILD_DIR:=$(BUILD_DIR)/libnvme-$(PKG_VERSION)
+PKG_HASH:=f73ba1edde059b2d5e7c1048ad4f895e6047bff241c94b34a7aff5894779d086
+
+PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
```
Comment (1715173329) [inline libs/libnvme/Makefile:17]:
```suggestion
PKG_LICENSE:=GPL-2.0-or-later
PKG_LICENSE_FILES:=LICENSE
PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
```

File: libs/libnvme/Makefile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+define Package/libnvme
+  TITLE:=libnvme
+  URL:=https://github.com/linux-nvme/libnvme
+  SECTION:=utils
+  CATEGORY:=Libraries
+  DEPENDS:=+libjson-c
+  SUBMENU:=Filesystem
+  LICENSE:=GPL-2.0-or-later
+  LICENSE_FILES:=LICENSE
```
Comment (1715173329) [inline libs/libnvme/Makefile:30]:
```suggestion
  SECTION:=utils
  CATEGORY:=Libraries
  SUBMENU:=Filesystem
  TITLE:=C Library for NVM Express on Linux
  URL:=https://github.com/linux-nvme/libnvme
  DEPENDS:=+libjson-c
```

sort, and convert LICENSE* to PKG_LICENSE*

File: libs/libnvme/Makefile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+endef
+
+define Package/libnvme/description
+NVM-Express C library.
+endef
```
Comment (1715173329) [inline libs/libnvme/Makefile:35]:
```suggestion
define Package/libnvme/description
  This is the libnvme development C library. libnvme provides type
  definitions for NVMe specification structures, enumerations, and
  bit fields, helper functions to construct, dispatch, and decode
  commands and payloads, and utilities to connect, scan, and manage
  nvme devices on a Linux system.
endef
```
from it's readme

File: utils/nvme-cli/Makefile [hunk 6]
```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2023 Luca Barbato
+
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
```
Comment (1715173329) [inline utils/nvme-cli/Makefile:6]: same here - seems unused

File: utils/nvme-cli/Makefile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+PKG_NAME:=nvme-cli
+PKG_VERSION:=2.5
+PKG_RELEASE:=1
+
+PKG_SOURCE:=nvme-cli-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/linux-nvme/$(PKG_NAME)/archive/refs/tags/v$(PKG_VERSION)
+PKG_BUILD_DIR:=$(BUILD_DIR)/nvme-cli-$(PKG_VERSION)
+PKG_HASH:=e84bdba276aadcddda8cf5d412e934cc5673af15132ea02180deb5d06af73146
```
Comment (1715173329) [inline utils/nvme-cli/Makefile:15]:
```suggestion
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/linux-nvme/nvme-cli/archive/refs/tags/v$(PKG_VERSION)
PKG_HASH:=e84bdba276aadcddda8cf5d412e934cc5673af15132ea02180deb5d06af73146
```

File: utils/nvme-cli/Makefile [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
```
Comment (1715173329) [inline utils/nvme-cli/Makefile:17]:
```suggestion
PKG_LICENSE:=GPL-2.0-or-later
PKG_LICENSE_FILES:=LICENSE
PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
```

File: utils/nvme-cli/Makefile [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+define Package/nvme-cli
+  TITLE:=nvme-cli
+  URL:=https://github.com/linux-nvme/nvme-cli
+  SECTION:=utils
+  CATEGORY:=Utilities
+  DEPENDS:=+libnvme 
+  SUBMENU:=Filesystem
+  LICENSE:=GPL-2.0-or-later
+  LICENSE_FILES:=LICENSE
```
Comment (1715173329) [inline utils/nvme-cli/Makefile:30]:
```suggestion
  SECTION:=utils
  CATEGORY:=Utilities
  SUBMENU:=Filesystem
  TITLE:=NVMe management command line interface
  URL:=https://nvmexpress.org/
  DEPENDS:=+libnvme 
```
sort, for URL, the repo description seems to point to https://nvmexpress.org/

File: utils/nvme-cli/Makefile [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+endef
+
+define Package/nvme-cli/description
+NVM-Express user space tooling for Linux.
```
Comment (1715173329) [inline utils/nvme-cli/Makefile:34]:
```suggestion
  NVM-Express user space tooling for Linux.
```

**Review by 1715173329 — CHANGES_REQUESTED** [review]

File: libs/libnvme/Makefile (hunk 2 above)
Comment (lu-zero) [reply libs/libnvme/Makefile:6]: You are right, I assumed it was needed but it is not.

**Comment by lu-zero** [comment]
Let me fold your suggestion with the removal of kernel.mk, thank you a lot for the review!

**Review by 1715173329 — APPROVED** [review]
LGTM - Thanks!

File: libs/libnvme/Makefile [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+PKG_LICENSE:=GPL-2.0-or-later
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/meson.mk
+
+define Package/libnvme
+  SECTION:=utils
```
Comment (jefferyto) [inline libs/libnvme/Makefile:23]:
```suggestion
  SECTION:=libs
```

Comment (lu-zero) [reply libs/libnvme/Makefile:23]: that's right ^^;

Comment (jefferyto) [reply libs/libnvme/Makefile:23]: Perhaps you can open a PR to fix this?

**Pull request merged**